packer/builder/azure/arm/artifact.go

```diff
@@ -3,34 +3,116 @@
 
 package arm
 
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
 const (
 	BuilderId = "Azure.ResourceManagement.VMImage"
 )
 
-type artifact struct {
-	name string
+type Artifact struct {
+	StorageAccountLocation      string
+	StorageAccountName          string
+	StorageAccountContainer     string
+	StorageAccountBlob          string
```

Comment:
I'd prefer 1 uri, called something like OSDiskUri, instead of account, container and blob.